web/src/message_lists.ts

```diff
@@ -52,10 +52,25 @@ export type MessageList = {
 export let home: MessageList | undefined;
 export let current: MessageList | undefined;
 
-export function set_current(msg_list: MessageList): void {
+function set_current(msg_list: MessageList): void {
+    // NOTE: Use update_current_message_list instead of this function.
     current = msg_list;
 }
 
+export function update_current_message_list(msg_list: MessageList): void {
+    if (msg_list !== home) {
+        home?.view.$list.removeClass("focused-message-list");
+    }
+
+    if (current !== home) {
+        // Remove old msg list from DOM.
+        current?.view.$list.remove();
+    }
+
+    set_current(msg_list);
+    current?.view.$list.addClass("focused-message-list");
+}
```

Comment:
Does this commit make any changes other than moving the code?

Reply:
Yeah, we use `update_current_message_list` in message_list.js instead of `set_current`.

Reply:
OK, and the only nontrivial thing that does is the last line of adding `focused-message-list` to `home_msg_list`, since `current` was previously unset. That might not be desirable, but I think isn't important?

Reply:
I just realized we don't need to call this anymore, so I removed `update_current_message_list` in message_list.js in the second commit since activate/deactivate will already call it. This seems like was a hack to keep `current` message list be always defined.